railties/lib/rails/commands/server/server_command.rb

```diff
@@ -10,6 +10,25 @@
 
 module Rails
   class Server < ::Rack::Server
+    module HandlerSuggestion
+      include Rails::Command::Behavior
+
+      # Hard-coding a bunch of handlers here as we don't have a public way of
+      # querying them from the Rack::Handler registry.
+      HANDLERS = %w[cgi fastcgi webrick lsws scgi thin puma unicorn]
+
+      def self.message(handler)
+        suggestions = HANDLERS
+          .sort_by { |h| levenshtein_distance(handler, h) }
+          .first(2)
+          .map! { |s| "'#{s}'" }
+
+        msg =  "Could not find handler '#{handler}'. ".dup
+        msg << "Maybe you meant #{suggestions.first} or #{suggestions.second}?\n"
```

Comment:
AFAICS this could end up suggesting exactly what they typed, which seems not-good. If `handler` is in `HANDLERS`, maybe we should instead suggest that they're missing a Gemfile entry.